### PR TITLE
better handle vegetation_type when vegetation_type is missing from grib file and also convert vegetation_type to string

### PR DESCRIPTION
**Pull request by randytpierce**
I discovered that I had not handled the case where a model didn't have any vegetation_type in the grib file, and i also realized that the handle_vegetation_type was not using the string value of veg_type in the json output. This PR corrects those oversights. I also adds a .prev and .tmp to the unit test for get_files.

### CODE REVIEW

**Review by randytpierce — COMMENTED** [review]
better handle vegetation_type

File: src/vxingest/grib2_to_cb/grib_builder_parent.py [hunk 1]
```diff
@@ -665,70 +665,62 @@ def build_document(self, queue_element):
                 self.ds_translate_item_variables_map = {
                     "2 metre temperature": ds_hgt_2_metre_temperature.variables[
                         list(ds_hgt_2_metre_temperature.data_vars.keys())[0]
-                        if len(list(ds_hgt_2_metre_temperature.data_vars.keys())) > 0
-                        else None
-                    ],
+                    ]
+                    if len(list(ds_hgt_2_metre_temperature.data_vars.keys())) > 0
+                    else None,
```
Comment (ian-noaa) [inline src/vxingest/grib2_to_cb/grib_builder_parent.py:670]: Ah, this is a nice catch. We want the map value equal to none (`ds_translate_item_variables_map["2 metre temperature"] == None`) when there aren't any keys in `ds_hgt_2_metre_temperature.data_vars`, correct?

File: tests/vxingest/netcdf_to_cb/test_unit_metar_obs_netcdf.py [hunk 2]
```diff
@@ -156,13 +156,20 @@ def test_vxingest_get_file_list(tmp_path):
...
         Path(tmp_path / "f_fred_02").touch()
         Path(tmp_path / "f_fred_04").touch()
         Path(tmp_path / "f_fred_05").touch()
         Path(tmp_path / "f_fred_03").touch()
         Path(tmp_path / "f_1_fred_01").touch()
+        Path(tmp_path / ".prev_fred").touch()
+        Path(tmp_path / ".prev/prev_fred").touch()
+        Path(tmp_path / ".tmp/tmp_fred").touch()
+        Path(tmp_path / "f_1_fred_01").touch()
         Path(tmp_path / "f_2_fred_01").touch()
```
Comment (ian-noaa) [inline tests/vxingest/netcdf_to_cb/test_unit_metar_obs_netcdf.py:173]: I like that we're testing for this now.

It may be good to start thinking about splitting this test up. It'd be good to have a test focused on making sure we get normal file lists back, and then another test to make sure we handle cases with special filenames, and then other test cases for other edge cases. That way if something does go wrong, we get more specific errors.

**Review by ian-noaa — COMMENTED** [review]
Generally looks okay. However, there was a comment that caused me some concern that I'd like to understand better. I'm hoping it's just leftover from development and can be deleted. 😄 

I also had a comment about splitting one of our tests out so each test is more focused and gives us a better indication of what went wrong. It's up to you if you want to do that now Randy. I understand you want to get this merged soon.

**Comment by ian-noaa** [comment]
Adding Gopa as a reviewer - I believe that was Randy's original intent.

File: src/vxingest/grib2_to_cb/grib_builder.py [hunk 3]
```diff
@@ -462,17 +476,46 @@ def handle_vegetation_type(self, params_dict):
...
             string: vegetation_type
         """
+        if self.ds_translate_item_variables_map["Vegetation Type"] is None:
+            return None
         values = self.ds_translate_item_variables_map["Vegetation Type"].values
         vegetation_type = []
+        # I don't know which land_use_type to use i.e. USGS,MODIFIED_IGBP_MODIS_NOAH,NLCD40,USGS-RUC, or MODI-RUC
+        # or which land_use_type_index i.e. "0"
+        land_use_type = "USGS"
+        land_use_type_index = "0"
```
[the remark this answers is not in the document]
Comment (randytpierce) [reply src/vxingest/grib2_to_cb/grib_builder.py:486]: > Is this comment still accurate? Did we resolve which of the `land_use_type`'s and `land_use_type_index`'s? If not, how did we decide on USGS?
> 
> I don't have a good feel for how significant getting this part correct is. But if there's doubt we're grabbing correct data, it might be better to figure that out before merging the PR.
> 
> If we did determine that USGS is the correct `land_use_type` to use, it'd be good to update the comment so we don't unintentionally give future readers doubt that the code is correct.

File: tests/vxingest/netcdf_to_cb/test_unit_metar_obs_netcdf.py (hunk 2 above)
Comment (randytpierce) [reply tests/vxingest/netcdf_to_cb/test_unit_metar_obs_netcdf.py:173]: I'm expecting to do that in the next PR, actually.

Comment (randytpierce) [reply tests/vxingest/netcdf_to_cb/test_unit_metar_obs_netcdf.py:173]: The next PR more fully addresses an error that in the get_file_list.

File: src/vxingest/grib2_to_cb/grib_builder.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (randytpierce) [reply src/vxingest/grib2_to_cb/grib_builder.py]: I modified the comment.

File: tests/vxingest/netcdf_to_cb/test_unit_metar_obs_netcdf.py (hunk 2 above)
Comment (ian-noaa) [reply tests/vxingest/netcdf_to_cb/test_unit_metar_obs_netcdf.py:173]: Sounds great!

File: src/vxingest/grib2_to_cb/grib_builder.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ian-noaa) [reply src/vxingest/grib2_to_cb/grib_builder.py]: Thanks, Randy!

**Review by ian-noaa — APPROVED** [review]

File: src/vxingest/grib2_to_cb/grib_builder_parent.py (hunk 1 above)
Comment (randytpierce) [reply src/vxingest/grib2_to_cb/grib_builder_parent.py:670]: yes
